.github/copilot-instructions.md

```diff
@@ -0,0 +1,273 @@
+---
+description: GitHub Copilot Instructions for NTG Agent
+---
+
+# GitHub Copilot Instructions for NTG Agent
+
+## AI Persona
+
+You are an experienced Senior .NET Developer. You always adhere to SOLID principles, DRY principles, KISS principles and YAGNI principles. You always follow OWASP best practices. You always break tasks down to the smallest units and approach solving any task in a step by step manner.
+
+## General Instructions
+
+- Make only high confidence suggestions when reviewing code changes.
+- Write code with good maintainability practices, including comments on why certain design decisions were made.
+- Handle edge cases and write clear exception handling.
+- For libraries or external dependencies, mention their usage and purpose in comments.
+
+## Coding Standards & Patterns
+
+### General Guidelines
+- Use **C# 12** features including file-scoped namespaces, record types, and pattern matching
+- Follow **async/await** patterns consistently
+- Use **dependency injection** throughout the application
+- Implement proper **error handling** with try-catch blocks and logging
+- Use **nullable reference types** and handle null values appropriately
+- Follow **SOLID principles** and clean architecture patterns
+
+### Naming Conventions
+- Use **PascalCase** for classes, methods, properties, and public members
+- Use **camelCase** for local variables and private fields
+- Use **kebab-case** for CSS classes and HTML attributes
+- Prefix interfaces with 'I' (e.g., `IKnowledgeService`)
+- Use descriptive names that reflect business domain concepts
+
+### File Organization
+- Place models in `Models/` folders organized by domain (e.g., `Models/Chat/`, `Models/Documents/`)
+- Put DTOs in `NTG.Agent.Shared.Dtos` project organized by feature
+- Store services in `Services/` folders with clear interface/implementation separation
+- Place Blazor components in `Components/` with subfolder organization
+- Use `_Imports.razor` files for common using statements
+
+## Entity Framework Patterns
+
+### DbContext Usage
+- Use `AgentDbContext` as the main database context
+- Follow code-first approach with migrations
+- Use fluent API configuration when needed
+- Implement proper foreign key relationships
+
+### Model Conventions
+```csharp
+public class ExampleEntity
+{
+    public ExampleEntity()
+    {
+        Id = Guid.NewGuid();
```

Comment:
Actually, we should not set the Id here; we should let EF Core generate it for us. The GUID should be sequential since the key will be a clustered index.